docs/sphinx/user_guide/feature/sort.rst

```diff
@@ -0,0 +1,141 @@
+.. ##
+.. ## Copyright (c) 2016-20, Lawrence Livermore National Security, LLC
+.. ## and other RAJA project contributors. See the RAJA/COPYRIGHT file
+.. ## for details.
+.. ##
+.. ## SPDX-License-Identifier: (BSD-3-Clause)
+.. ##
+
+.. _sort-label:
+
+================
+Sorts
+================
+
+RAJA provides portable parallel sort operations, which are basic
+parallel algorithm building blocks. They are described in this section.
+
+A few important notes:
+
+.. note:: * All RAJA sort operations are in the namespace ``RAJA``.
+          * Each RAJA sort operation is a template on an *execution policy*
+            parameter. The same policy types used for ``RAJA::forall`` methods
+            may be used for RAJA sorts.
+          * RAJA sort operations accept an optional *comparator* argument so
+            users can perform different types of sort operations. If
+            no operator is given, the default is a 'less' operation and
+            the result is **non-decreasing**.
+
+Also:
+
+.. note:: * For sorts using the CUDA back-end, RAJA uses the implementations
+            provided by the NVIDIA cub library, which is available in the
+            RAJA source repository as a Git submodule. The CMake variable
+            ``CUB_DIR`` will be automatically set to the location of the cub
+            library when CUDA is enabled; to use a different version of the
+            cub library, install it and set the ``CUB_DIR`` variable to the
+            desired location when running CMake.
+          * The RAJA CUDA back-end implementation only supports sorting
+            arithmetic types using RAJA operators less and greater.
+
+Please see the :ref:`sort-label` tutorial section for usage examples of RAJA
+sort operations.
+
+-----------------
+Sort Operations
+-----------------
+
+In general, a sort operation takes a sequence of numbers 'x' and a binary
+comparison operator 'op' that forms a strict weak ordering of the elements in x
+as input and produces a sequence of numbers 'y' as output. The output sequence
+is a permutation of the input sequence where each pair of elements 'a' and 'b,'
+where a is before b in the output sequence, satisfies !(b op a).
+Sorts are stable if they always preserve the order of equivalent elements,
+where equivalent elements satisfy !(a op b) && !(b op a).
```

Comment:
The '&&' looks a little finny in the generated docs. Better to replace with 'and'?

Reply:
Again, I'd use verbatim formatting here, or if it's a  mathematical/logical equation then math formatting. Leaving it formatted like the surrounding normal text makes the operators look very peculiar.  The convention we use for things like this in the OpenMP spec is to verbatim format short code blocks like that, then italicise placeholders like *op* inside it to distinguish them from literal code.